### ISSUE TITLE
Map ref inconsistency after 'Edit record'

### ISSUE THREAD

**Issue by DavidHepper** [BiologicalRecordsCentre/iRecord#1216]
Record ID 23054696 has had the map reference edited. It now shows NM91310195 in the grid but the new (corrected) NM917020 in 'Edit record'. 
This record updated unexpectedly in 'Edit record', probably when I pressed Enter in the 'Spatial Ref' field but I don't know whether that was the cause. I would not expect Enter in a form field to submit changes but only to accept that field.

**Comment by kitenetter** [comment]
The record is correctly stored in the warehouse, but the output map reference has not updated, at least in the ElasticSearch view. This leads to an incorrect output map reference in the download file.

It has always been the case that pressing Enter triggers the record to save.

**Comment by johnvanbreda** [comment]
I've fixed a bug where the enter key in the grid ref field wasn't updating the saved geometry as the form was saving first. This was originally changed to avoid a bug in IE, so I've reverted it as we can't continue supporting an unsupported browser.

I've repaired this specific record and scanned other data as best I can but could not find any other similar problems - if you find any let me know.
